### PR TITLE
code refactoring

### PR DESCRIPTION
**Pull request by Zain-ul-din**
added type to `config`.

### CODE REVIEW

**Review by Zain-ul-din — COMMENTED** [review]
tested by my side

**Review by Zain-ul-din — COMMENTED** [review]
done

**Comment by Zain-ul-din** [comment]
@navopw please can you merge and fix this?

File: src/config.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+// Environment variables
+import dotenv from "dotenv";
+dotenv.config();
+
+const config:ConfigType = {
+	prefixEnabled: process.env.PREFIX_ENABLED == "true",
+	gptPrefix: "!gpt",
+	dallePrefix: "!dalle"
+}
```
Comment (marcomafessolli) [inline src/config.ts:11]: Why not just declare ConfigType here? -1 file

File: src/helper/CONSTANT.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export enum CONSTANT  {
+    StatusBroadcast = "status@broadcast"
+};
```
Comment (marcomafessolli) [inline src/helper/CONSTANT.ts:3]: I'd move this to the `config.ts` file as well and use it as const for better intellesence. -1 file

```suggestion
export const constants =  {
    StatusBroadcast = "status@broadcast"
} as const;
```

**Comment by Zain-ul-din** [comment]
> Did the refactoring already:
> 
> image

It seems that the main branch in @navopw doesn't match what was written, which has invalidated my efforts. Given the behavior of the authors, I don't think it's reasonable to continue any further. Good luck with your repository.

**Comment by navopw** [comment]
Hey @Zain-ul-din, sad to hear that. I didn't want to invalidate your efforts.

I was waiting for the changes from the review and quickly did it myself (5 minutes work). Really like to proceed asap.

I have adopted your changes almost exactly:
- Constants
- Config

What I haven't adopted:
- Whatsapp & messages functions in separate file

I also wish you the best!

**Comment by Zain-ul-din** [comment]
@navopw 

image
SS from master can yuo compare my index.ts with it?

**Comment by navopw** [comment]
Yours is different since you moved out the sendMessage logic (which I renamed to handleIncomingMessage).

Also like I stated:
```
What I haven't adopted:

Whatsapp & messages functions in separate file
```

**Comment by Zain-ul-din** [comment]
The function `handleIncomingMessage` is located in a distinct file within the pull request. Do you recall agreeing to merge the pull request and simply include config.ts? However, you did not follow through on your commitment.

**Comment by navopw** [comment]
Feel free to contribute another PR that does this. I will merge it.

Don't have the time to discuss such small details.

However it's a good idea to put it in a separate file.

**Pull request closed without merging**